lib/smart_answer/title.rb

```diff
@@ -0,0 +1,29 @@
+module SmartAnswer
+  class Title
+    def initialize(content)
+      @content = content
+    end
+
+    def wrapped_with_debug_div?
+      @content.match(/^<div data-debug-template-path=.*><p>.*<\/p><\/div>$/i)
+    end
+
+    def text
+      return @content unless wrapped_with_debug_div?
+
+      document_fragment.children.first.children.first.children.text
+    end
+
+    def partial_template_path
+      return unless wrapped_with_debug_div?
+
+      document_fragment.children.first.attributes.first.last.value
+    end
+
+  private
+
+    def document_fragment
+      @document_fragment ||= Nokogiri::HTML::DocumentFragment.parse(@content)
+    end
+  end
+end
```

Comment:
The logic in this class seems to be exclusively view/presentation-related and as such I think this should be a presenter/renderer class rather than what looks to be more of a "model" class at the moment.